src/code/z_room.c

```diff
@@ -614,49 +641,56 @@ s32 func_8009728C(PlayState* play, RoomContext* roomCtx, s32 roomNum) {
         roomCtx->curRoom.segment = NULL;
         roomCtx->status = 1;
 
-        ASSERT(roomNum < play->numRooms, "read_room_ID < game_play->room_rom_address.num", "../z_room.c", 1009);
+        ASSERT(roomNum < play->roomList.count, "read_room_ID < game_play->room_rom_address.num", "../z_room.c", 1009);
 
-        size = play->roomList[roomNum].vromEnd - play->roomList[roomNum].vromStart;
-        roomCtx->unk_34 =
-            (void*)ALIGN16((uintptr_t)roomCtx->bufPtrs[roomCtx->unk_30] - ((size + 8) * roomCtx->unk_30 + 7));
+        size = play->roomList.romFiles[roomNum].vromEnd - play->roomList.romFiles[roomNum].vromStart;
+        roomCtx->roomRequestAddr = (void*)ALIGN16((uintptr_t)roomCtx->bufPtrs[roomCtx->activeBufPage] -
+                                                  ((size + 8) * roomCtx->activeBufPage + 7));
 
         osCreateMesgQueue(&roomCtx->loadQueue, &roomCtx->loadMsg, 1);
 
 #if PLATFORM_N64
         if ((B_80121220 != NULL) && (B_80121220->unk_08 != NULL)) {
             B_80121220->unk_08(play, roomCtx, roomNum);
         } else {
-            DMA_REQUEST_ASYNC(&roomCtx->dmaRequest, roomCtx->unk_34, play->roomList[roomNum].vromStart, size, 0,
-                              &roomCtx->loadQueue, NULL, "../z_room.c", 1036);
+            DMA_REQUEST_ASYNC(&roomCtx->dmaRequest, roomCtx->roomRequestAddr,
+                              play->roomList.romFiles[roomNum].vromStart, size, 0, &roomCtx->loadQueue, NULL,
+                              "../z_room.c", 1036);
         }
 #else
-        DMA_REQUEST_ASYNC(&roomCtx->dmaRequest, roomCtx->unk_34, play->roomList[roomNum].vromStart, size, 0,
-                          &roomCtx->loadQueue, NULL, "../z_room.c", 1036);
+        DMA_REQUEST_ASYNC(&roomCtx->dmaRequest, roomCtx->roomRequestAddr, play->roomList.romFiles[roomNum].vromStart,
+                          size, 0, &roomCtx->loadQueue, NULL, "../z_room.c", 1036);
 #endif
 
-        roomCtx->unk_30 ^= 1;
+        roomCtx->activeBufPage ^= 1;
         return true;
     }
 
     return false;
 }
 
-s32 func_800973FC(PlayState* play, RoomContext* roomCtx) {
+/**
+ * Completes room initialization for the requested room by calling Room_RequestNewRoom.
```

Comment:
this comment is not true.

Reply:
it's worded confusingly, I suggested a rewording

Reply:
it's not confusing, it's just a lie. the function does not call `Room_RequestNewRoom`

Reply:
https://github.com/zeldaret/oot/pull/2015#discussion_r1744492815

Reply:
oh, sniped!